WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -1,47 +1,44 @@
 import Foundation
 
 
-
-// MARK: - ManagedObject Base Class
+// MARK: - NSManagedObject Default entityName Helper
 //
-protocol ManagedObject {
-    static var entityName: String { get }
+extension NSManagedObject {
+    class var entityName: String {
+        return classNameWithoutNamespaces()
+    }
 }
 
 
-// MARK: - Core Data Helpers
+// MARK: - NSManagedObjectContext Helpers!
 //
-struct CoreDataHelper<T> where T: NSManagedObject, T: ManagedObject {
-    /// CoreData ManagedObjectContext
-    ///
-    let context: NSManagedObjectContext
-
+extension NSManagedObjectContext {
 
     /// Returns a new FetchRequest for the associated Entity
     ///
-    func newFetchRequest() -> NSFetchRequest<NSFetchRequestResult> {
-        return NSFetchRequest(entityName: T.entityName)
+    func newFetchRequest<T: NSManagedObject>(forType type: T.Type) -> NSFetchRequest<NSFetchRequestResult> {
+        return NSFetchRequest(entityName: type.entityName)
     }
 
-
     /// Returns all of the entities that match with a given predicate.
     ///
     /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
     ///
-    func allObjects(matchingPredicate predicate: NSPredicate? = nil, sortedBy descriptors: [NSSortDescriptor]? = nil)  -> [T] {
-        let request = newFetchRequest()
+    func allObjects<T: NSManagedObject>(ofType type: T.Type, matching predicate: NSPredicate? = nil, sortedBy descriptors: [NSSortDescriptor]? = nil) -> [T] {
+        let request = newFetchRequest(forType: type)
         request.predicate = predicate
         request.sortDescriptors = descriptors
 
-        return loadObjects(withFetchRequest: request)
+        return loadObjects(ofType: type, with: request)
     }
 
+
     /// Returns the number of entities found that match with a given predicate.
     ///
     /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
     ///
-    func countObjects(matchingPredicate predicate: NSPredicate? = nil) -> Int {
-        let request = newFetchRequest()
+    func countObjects<T: NSManagedObject>(ofType type: T.Type, matching predicate: NSPredicate? = nil) -> Int {
+        let request = newFetchRequest(forType: type)
         request.predicate = predicate
         request.includesSubentities = false
         request.predicate = predicate
```

Comment:
Not something you introduced in this PR but should get fixed - setting predicate twice.